### ISSUE TITLE
Unable to perform Git Push in WSL 2

### ISSUE THREAD

**Issue by jherring00** [microsoft/WSL#7302]
### Windows Build Number

Microsoft Windows [Version 10.0.19043.1165]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.4.72

### Distro Version

Ubuntu 18.04

### Other Software

Git version 2.17.1

### Repro Steps

1. Upgrading from WSL 1 to WSL 2
2. Init a new Git Repo / Clone a Git Repo
3. Add changed files to commit
4. Make a commit
5. Push commit to GitHub

### Expected Behavior

Expecting a normal push that completes successfully.

### Actual Behavior

Instead of the normal behaviour expected, the following occurs immediately with no hanging.
Running git push in WSL 1 works successfully but does not work immediately after upgrading Ubuntu 18.04 to WSL 2
```
Writing objects: 100% (16/16), 3.31 KiB | 1.65 MiB/s, done.
Total 16 (delta 5), reused 0 (delta 0)
Connection to github.com closed by remote host.
fatal: The remote end hung up unexpectedly
fatal: The remote end hung up unexpectedly
```

**Comment by benhillis** [comment]
I'm not able to reproduce this, I know GitHub was having some issues yesterday, can you still repro today?

**Comment by jherring00** [comment]
> I'm not able to reproduce this, I know GitHub was having some issues yesterday, can you still repro today?

Yes, unfortunately I still received the same exact error as found in the Original Post. This error occurred prior to the issues they had yesterday and I believe it's an independent issue. My thoughts lean towards it being a networking issue with WSL 2 but I'm not experienced enough to diagnose the issue further.

**Comment by craigloewen-msft** [comment]
Are you able to connect to the internet normally? Are you using any form of VPN software on this machine? I'm also not able to reproduce this either so we'd likely need more info to determine what's going on.

**Comment by jherring00** [comment]
> Are you able to connect to the internet normally? Are you using any form of VPN software on this machine? I'm also not able to reproduce this either so we'd likely need more info to determine what's going on.

I am able to use the internet normally. The issue with git push exists exclusively within WSL 2. Downgrading to WSL 1 removes the issue entirely. I do not have any VPN software on the machine. I understand this isn't a whole lot to go off of but I would certainly appreciate any tips on diagnosing the issue further.

**Comment by jherring00** [comment]
I have resolved the issue by reinstalling the WSL 2 Linux kernel update package as dictated in [Step 4 of the WSL installation guide.](https://docs.microsoft.com/en-us/windows/wsl/install-win10)

**Comment by tonyadolph** [comment]
With WSL2 Ubuntu 20.04.5 LTS
I could not ssh -T git@github.com until I set HostKeyAlgorithms in /etc/ssh/ssh_config as follows:

`HostKeyAlgorithms ecdsa-sha2-nistp256`

Then I was getting the errors reported here:

`Enumerating objects: 59, done.
Counting objects: 100% (59/59), done.
Delta compression using up to 12 threads
Compressing objects: 100% (55/55), done.
Writing objects: 100% (59/59), 1000.76 KiB | 3.83 MiB/s, done.
Total 59 (delta 10), reused 0 (delta 0)
fatal: the remote end hung up unexpectedly
fatal: the remote end hung up unexpectedly
Connection to github.com closed by remote host.`


I tried many buffer size changes etc.    

But the fix was to change my Wifi adapter - **thanks** [jherring00](https://github.com/jherring00)
